### ISSUE TITLE
False positive for no-lonely-if with a nested conditional

### ISSUE THREAD

**Issue by yaycmyk** [eslint/eslint#3722]
Sample code:

```js
xhr({uri: '/services/services_fixture.json'}, function handleResponse(error, response, body) {
    if (error) {
        console.error(error.message);
        reject();
    } else {
        if (response.statusCode > 199 && response.statusCode < 400) {
            try {
                this.services = JSON.parse(body);
                resolve(this.services);
            } catch (e) {
                console.error(e.message);
                reject();
            }
        } else {
            reject();
        }
    }

    this.fetchPromise = null;
});
```

I'm getting an error for the `if` statement inside the first `else` block, when there clearly is an `else` for that `if` as well.

This is on version `1.3.1`, with `"no-lonely-if": 2`

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
Can you please share the exact output of eslint?
What kind of parser are you using?

**Comment by michaelficarra** [comment]
@yaycmyk You're still violating the rule. Your first `else` should be an `else if`.

**Comment by yaycmyk** [comment]
@michaelficarra The rule text should be extended to say that `if``else` that can be hoisted will trigger this rule then. Currently it only mentions an `if` by itself inside another `if` and nothing about an `if``else` pair.

**Comment by IanVS** [comment]
Wouldn't hurt to add an example to the docs, IMO.

**Comment by michaelficarra** [comment]
Have any of you looked at the documentation? Please just look at it. http://eslint.org/docs/rules/no-lonely-if The entirety of the documentation for this rule talks specifically about this case.

**Comment by IanVS** [comment]
Yes I looked at it.  Existing does are:

```js
/*eslint no-lonely-if: 2*/

if (condition) {
    // ...
} else {
    if (anotherCondition) { /*error Unexpected if as the only statement in an else block.*/
        // ...
    }
}
```

I can see where the message could be somewhat misleading (`Unexpected if as the only statement in an else block.`), and since this has tripped up at least one person, I'm saying it wouldn't hurt to also include another `else`:

```js
if (condition) {
    // ...
} else {
    if (anotherCondition) { /*error Unexpected if as the only statement in an else block.*/
        // ...
    } else {
        // ...
    }
}
```

**Comment by yaycmyk** [comment]
@michaelficarra That's a pretty condescending statement considering I very clearly stated in the previous message that there was no specific info regarding a nested `if else` block versus a singular nested `if`.

I agree with @IanVS that adding an example or just a disclaimer line to the docs would be enough.

**Comment by michaelficarra** [comment]
@yaycmyk My apologies, I didn't realise you were considering an if without an else as a different thing than an if with an else. To me, `if (a) b;` is the same thing as `if (a) b; else ;`. Additionally, they are both represented by the same AST node, so those structures are quite tightly associated in my mind. If changing one of the examples to include an else helps someone, I'm not opposed.

**Comment by yaycmyk** [comment]
:+1:
